packages/xo-web/src/xo-app/vm/tab-disks.js

```diff
@@ -833,52 +807,49 @@ export default class TabDisks extends Component {
       sr.$container === requiredHost
   )
 
-  _getVbdsByVdi = createSelector(
-    () => this.props.vdis,
+  _getVbds = createSelector(
     () => this.props.vbds,
-    () => this.props.vm,
-    (vdis, vbds, vm) => mapValues(vdis, vdi => find(vbds, { VDI: vdi.id }))
-  )
-
-  _getIsVdiAttached = createSelector(
-    createSelector(
-      () => this.props.allVbds,
-      () => Object.keys(this.props.vdis),
-      (vbds, vdis) => pick(groupBy(vbds, 'VDI'), vdis)
-    ),
-    vbdsByVdi => mapValues(vbdsByVdi, vbds => some(vbds, 'attached'))
+    () => this.props.vdis,
+    () => this.props.srs,
+    (vbds, vdis, srs) =>
+      compact(
+        map(vbds, vbd => {
+          const vdi = vdis[vbd.VDI]
+          return vdi !== undefined && { ...vbd, vdi, vdiSr: srs[vdi.$SR] }
```

Comment:
Should we show VBDs with no VDI attached? What does it mean? Maybe in another PR?

Reply:
As discussed, I'll do it in another PR.